spec/lib/approval_request_listener_spec.rb

```diff
@@ -0,0 +1,111 @@
+describe ApprovalRequestListener do
+  let(:approval_request_ref) { "0" }
+  let(:client)               { double(:client) }
+  let(:event)                { ApprovalRequestListener::EVENT_REQUEST_FINISHED }
+  let(:payload)              { {"request_id" => request_id, "decision" => decision, "reason" => reason } }
+  let(:request_id)           { "1" }
+
+  describe "#subscribe_to_approval_updates" do
+    let(:request) do
+      { :headers => { 'x-rh-identity' => encoded_user_hash }, :original_url => 'whatever' }
+    end
+
+    around do |example|
+      ManageIQ::API::Common::Request.with_request(request) { example.call }
```

Comment:
@syncrou This spec demonstrates that we need headers to locate our objects. @Fryguy mentioned that we have to pass headers back in the Kafka messages so we can locate our objects.

Reply:
@mkanoor are you saying we should publish messages back to Kafka from the Approval listener? I thought the discussion was that we were going to receive that information back in the approval published events payload.

Reply:
@syncrou Its what we receive back in the messages we are already sending the headers to Approval Service as part of @lindgrenj6 PR #167

Reply:
I understand that when we are making REST calls to other API that we forward headers, this makes sense as the api calls could be internal or external callers in most cases and would present user/tenant data consistently.

I think it is wrong to say that every kafka message should include the same thing.  The message queue is internal only and passing and saving headers feels problematic.  Are we really just saying that the kafka message needs to supply the tenant ID so we can lookup the resources?

Approval already knows the tenant the request was created for an can easily add that to the message.

cc @bzwei

Reply:
I lean to the idea of  passing around the tenant id only, rather than the whole header.